src/picture.rs

```diff
@@ -121,20 +126,23 @@ pub unsafe extern "C" fn dav1d_default_picture_alloc(
 
     (*p_c).stride = stride;
     (*p_c).data = data.map(NonNull::new);
-    (*p_c).allocator_data = NonNull::new(buf.cast());
+    (*p_c).allocator_data = NonNull::new(Box::into_raw(buf).cast::<c_void>());
     // The caller will create the real `Rav1dPicture` from the `Dav1dPicture` fields set above,
     // so we don't want to drop the `Rav1dPicture` we created for convenience here.
     mem::forget(p);
 
     Rav1dResult::Ok(()).into()
 }
 
-pub unsafe extern "C" fn dav1d_default_picture_release(p: *mut Dav1dPicture, cookie: *mut c_void) {
-    rav1d_mem_pool_push(
-        cookie as *mut Rav1dMemPool,
-        (*p).allocator_data
-            .map_or_else(ptr::null_mut, |data| data.as_ptr()) as *mut Rav1dMemPoolBuffer,
-    );
+pub unsafe extern "C" fn dav1d_default_picture_release(p: *mut Dav1dPicture, _cookie: *mut c_void) {
+    let buf = (*p)
+        .allocator_data
+        .unwrap()
+        .as_ptr()
+        .cast::<MemPoolBuf<u8>>();
```

Comment:
Should this be `MemPoolBuf<MaybeUninit<u8>>` now?

Reply:
Yup.  Missed that one.